src/control/server/engine/tags.go

```diff
@@ -155,7 +155,9 @@ func parseCmdTags(in interface{}, tagFilter string, joiner joinFn, seenRefs refM
 				if fVal.Len() == 0 && opts.hasOpt(nonZero) {
 					continue
 				}
-				strVal := strconv.Itoa(fVal.Len())
```

Comment:
I remember now why/when this was added... The storage tiering work added it to support a new `-T` flag to tell the engine how many storage tiers it should expect: https://github.com/daos-stack/daos/blob/master/src/control/server/storage/config.go#L572

I think in order to support both use cases you'll just have to add a new directive (like `intBool` or `invertBool`) to tell the parser how to interpret a slice. I'm fine with making this new implementation the default, as the other one is kind of strange. Maybe call it `sliceCount` ?

Reply:
Good catch, I've fixed it.